08_project/alextmz/pkg/puppy/store/mapstore.go

```diff
@@ -0,0 +1,56 @@
+package store
+
+import "github.com/anz-bank/go-course/08_project/alextmz/pkg/puppy"
+
+type MapStore struct {
+	pmap   map[int]puppy.Puppy
+	nextID int
+}
+
+func NewMapStore() MapStore {
+	a := MapStore{pmap: make(map[int]puppy.Puppy)}
+	return a
+}
+
+func (m MapStore) CreatePuppy(p *puppy.Puppy) error {
```

Comment:
You should fix it up here... needs to be a bpointer receiver.
Also: write a test, e.g. create & read that fails with this version and works with pointer receivers.

Reply:
fixed